src/stories/Tooltip.stories.tsx

```diff
@@ -22,10 +23,185 @@ export default {
   ],
 } as Meta
 
-export const IconButtonTooltip = () => (
+export const CurrentTooltip = () => (
   <Box sx={{p: 5}}>
-    <Tooltip aria-label="Search">
+    <Tooltip text="Search">
       <IconButton icon={SearchIcon} aria-label="Search" />
     </Tooltip>
   </Box>
 )
+
+// As a label for an IconButton
+export const Tooltip2LabelTypeTooltip = () => (
+  <Box sx={{p: 5}}>
+    <Tooltip2 aria-label="Search tooltip">
+      <IconButton icon={SearchIcon} aria-label="Search button itself" />
+    </Tooltip2>
+  </Box>
+)
+
+// As a label for an IconButton
+export const Tooltip2CustomIcon = () => (
```

Comment:
For some reason this is getting a little cut off. I'm not sure if you worked on viewport boundaries at all, but I would assume it should always reposition itself to not break the viewport

<img width="262" alt="image" src="https://user-images.githubusercontent.com/18661030/227077033-78f8237e-ae11-4f88-9931-4d4a84953606.png">

Reply:
As far as I understand, this would be achieved with the popover api ([issue](https://github.com/primer/react/issues/2124) for ref) and unfortunately this tooltip is still uses `z-index`. This PR is mainly remediating the existing Tooltip a11y issues and I thought it would be a good opportunity to update the styles as well for saving us time in the near future when migrating them over to .css

Reply:
@langermank I will see how much effort needs for this to make it with the current structure. If that is okay, I'd like to tackle this separately, independent from this PR.